### PR TITLE
Js unit testing

### PR DESCRIPTION
**Pull request by tchalvak**
Get initial Js tests running for CI, and for make test-js

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -86,3 +86,5 @@ Icon
 .TemporaryItems
 .Trashes
 .VolumeIcon.icns
+node_modules/
+node_modules/
```
Comment (Beagle) [inline .gitignore]: why twice?
why twice?

File: deploy/tests/karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    // list of files / patterns to load in the browser
+    files: [
+      'test-main.js',
+      'deploy/www/js/jquery.min.js',
+      'deploy/www/js/jquery.timeago.js',
+      'deploy/www/js/nw.js',
+      'deploy/www/js/debug.js',
+      'deploy/www/js/passwords.js',
+      'deploy/www/js/talk.js',
```
Comment (Beagle) [inline deploy/tests/karma.conf.js]: can you make "deploy/www/js" a variable so as to not be so repetitive with hard-coded paths?

**Comment by Beagle** [comment]
Besides my picked nits, :shipit: merge at your leisure

File: deploy/tests/karma.conf.js (hunk 2 above)
Comment (tchalvak) [inline deploy/tests/karma.conf.js]: Maybe later.

File: .gitignore (hunk 1 above)
Comment (tchalvak) [inline .gitignore]: Just echoed it twice accidentally apparently.  Will remove one at some point.

**Pull request merged**